Data mock for Goal page

celinevu opened the pull request:
## Summary

What is the nature of this pull request?

- [ x] New feature
- [ ] Enhancement to an existing feature
- [ ] Bug fix
- [ ] Configuration update
- [ ] Dependency update

### Description of change
Please replace this text with a description of your change.

### Planner card link
[Replace this with your task link](#url)

## Readiness

- [ ] The application runs successfully with my change.
- [ ] I have included [code comments](https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments/) where appropriate.
- [ ] Where I have copied and pasted code from other sources, I have included a comment with a link to the original source.
- [ ] I have read and followed the [principles and guidelines about submitting code](https://redback-operations.github.io/redback-documentation/docs/web-mobile-app-dev/frontend/submitting-work).
- [ ] I have added [unit tests](https://redback-operations.github.io/redback-documentation/docs/web-mobile-app-dev/frontend/tests) for new components and/or functionality.
- [ ] All existing and new unit tests pass (including updating tests for any necessary breaking changes).

celinevu committed:
Data mock for Goal page

MorgaineBarter commented:
this PR appears to have a lot of 'minor spacing/format changes' contained within. 
if your vs code has automatically 'reformat on save' turned on - turn it off asap!
this makes it difficult to see the actual changes made.

celinevu committed:
Firebase project for login/sign up authentication
-	Installed and integrated required packages (firebase, @react-firebase/auth, axios).
-	Ensured proper environment variable handling for secure API key management.
-	Enabled Google Sign-In as an authentication provider in the Firebase Co

celinevu committed:
Friends Button Feature Development:
Fetch functions to handle friend request actions (Accept, Decline).

Narendran99 committed:
Merge branch 'main' into main

Narendran99 reviewed (approved):
I've resolved a couple of minor issues. No other conflicts. Approving the PR.
